### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by albertoperdomo**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+METABASE_SITE_URL = ENV.fetch('METABASE_SITE_URL') || 'http://localhost:3000'
+METABASE_EMBED_DASHBOARD_ID= ENV.fetch('METABASE_EMBED_DASHBOARD_ID')
+
+get '/' do
+  payload = {
+    :resource => {:dashboard => 8},
+    :params => {          
+    },
+    :exp => Time.now.to_i + (60 * 10) # 10 minute expiration
+  }
```
Comment (rdejuana) [inline app.rb:13]: if you are going to use symbols in a hash, the style is not to use the =>.  So more like
```
payload = {
    resource: { dashboard: 8},
    params:  {},
    exp: Time.now.to_i + (60 * 10) # 10 minute expiration
  }
```

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'sinatra'
+require 'jwt'
+
+METABASE_EMBEDDING_SECRET = ENV.fetch('METABASE_EMBEDDING_SECRET')
+METABASE_SITE_URL = ENV.fetch('METABASE_SITE_URL') || 'http://localhost:3000'
+METABASE_EMBED_DASHBOARD_ID= ENV.fetch('METABASE_EMBED_DASHBOARD_ID')
```
Comment (rdejuana) [inline app.rb]: you're not aren't using this in the code.  What is it for?

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+get '/' do
+  payload = {
+    :resource => {:dashboard => 8},
+    :params => {          
+    },
+    :exp => Time.now.to_i + (60 * 10) # 10 minute expiration
+  }
+  token = JWT.encode payload, METABASE_EMBEDDING_SECRET
+  iframe_url = METABASE_SITE_URL + "/embed/dashboard/" + token + "#bordered=true&titled=true"
```
Comment (rdejuana) [inline app.rb]: I'm fifty fifty on whether to use string interpolation or concatenation .

Should the `#` be a `?`? Not really sure what happens to the `titled` param.  If it comes after an anchor.

File: app.rb (hunk 2 above)
Comment (albertoperdomo) [reply app.rb]: That was an oversight. I should be using that for the resource ID in the JWT payload.

File: app.rb (hunk 3 above)
Comment (albertoperdomo) [reply app.rb]: >I'm fifty fifty on whether to use string interpolation or concatenation .

To be fair, I am not a fan of neither of the two. I think any of the two ways to construct the URL is brittle.
For instance, if there is a trailing `/` in `METABASE_SITE_URL`, then there will be a double `//` in the URL.

I will explore using the URI module to construct the URL instead.

File: app.rb (hunk 1 above)
Comment (albertoperdomo) [reply app.rb:13]: Updated.

**Comment by albertoperdomo** [comment]
Thanks so much for the review @rdejuana ! 🙇

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+require 'sinatra'
+require 'jwt'
+
+METABASE_EMBEDDING_SECRET = ENV.fetch('METABASE_EMBEDDING_SECRET')
+METABASE_SITE_URL = ENV.fetch('METABASE_SITE_URL') || 'http://localhost:3000'
+METABASE_EMBED_DASHBOARD_ID= ENV.fetch('METABASE_EMBED_DASHBOARD_ID').to_i
```
Comment (NevRA) [inline app.rb]: Nit: whitespace

**Review by NevRA — APPROVED** [review]

**Pull request merged**